src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -152,6 +154,55 @@ internal JsonClassInfo(Type type, JsonSerializerOptions options)
                         ElementClassInfo = options.GetOrAddClass(elementType);
                     }
                     break;
+                case ClassType.KeyValuePair:
+                    {
+                        // For KeyValuePair, we would ordinarily treat it as ClassType.Object for both serialization and deserialization
```

Comment:
After the converters are in, I think we can treat this as a normal converter that will call back into the serializer (Read or Write methods) to read\write the object side.